executor/substate_provider.go

```diff
@@ -31,13 +33,21 @@ import (
 // ----------------------------------------------------------------------------
 
 // OpenSubstateProvider opens a substate database as configured in the given parameters.
-func OpenSubstateProvider(cfg *utils.Config, ctxt *cli.Context, aidaDb db.BaseDB) Provider[txcontext.TxContext] {
+func OpenSubstateProvider(cfg *utils.Config, ctxt *cli.Context, aidaDb db.BaseDB) (Provider[txcontext.TxContext], error) {
 	substateDb := db.MakeDefaultSubstateDBFromBaseDB(aidaDb)
+
+	if cfg.SubstateEncoding != "" {
+		_, err := substateDb.SetSubstateEncoding(cfg.SubstateEncoding)
+		if err != nil {
+			return nil, fmt.Errorf("Fail to set substate encoding; %w", err)
```

Comment:
errors in Go should be lower case

Reply:
Fixed
[842a212](https://github.com/Fantom-foundation/Aida/pull/1170/commits/842a212a3be4247eb37cd7f1b0615087e160fedc)